lib/Doctrine/Common/Annotations/AnnotationRegistry.php

```diff
@@ -110,6 +126,9 @@ static public function registerLoader($callable)
         if (!is_callable($callable)) {
             throw new \InvalidArgumentException("A callable is expected in AnnotationRegistry::registerLoader().");
         }
+        // Reset our static cache now that we have a new loader to work with
+        self::$loaded = array();
```

Comment:
Loaded classes do not need to be reset - there is nothing that we can reset anyway